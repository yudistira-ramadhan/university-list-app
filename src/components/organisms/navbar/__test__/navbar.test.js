import React from 'react'
import ReactDOM from 'react-dom'
import Navbar from '../navbar'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Navbar/>, div)
})

it('renders Navbar correctly', () => {
  const { getByTestId } = render(<Navbar />)
  expect(getByTestId('navbar')).toBeTruthy()
})

it('matches snapshot', () => {
  const tree = renderer.create(<Navbar />).toJSON()
  expect(tree).toMatchSnapshot()
})