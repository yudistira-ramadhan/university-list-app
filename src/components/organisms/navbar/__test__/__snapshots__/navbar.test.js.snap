// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-info"
  data-testid="navbar"
>
  <a
    className="font-weight-bold navbar-brand"
    href="/"
  >
    University List App
  </a>
  <button
    aria-label="Toggle navigation"
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="mt-2 navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="font-weight-bold nav-link"
          href="/country"
          onClick={[Function]}
        >
          List by Country
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="font-weight-bold nav-link"
          href="/name"
          onClick={[Function]}
        >
          List by Name
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="font-weight-bold nav-link"
          href="/detail"
          onClick={[Function]}
        >
          University Detail
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
